v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -358,7 +390,7 @@ async function parseNamedParameters(
         (cliArguments[i + 1] === "true" || cliArguments[i + 1] === "false")
```

Comment:
it doesn't let me comment on the right line, but the expression could be simplified as `usedCliArguments[i + 1] === false && (cliArguments[i + 1] === "true" || cliArguments[i + 1] === "false")`

Reply:
N/A, see https://github.com/NomicFoundation/hardhat/pull/5334#discussion_r1628104331